src/shogun/base/SGObject.cpp

```diff
@@ -471,6 +473,22 @@ void CSGObject::save_serializable_post() throw (ShogunException)
 	m_save_post_called = true;
 }
 
+template<class Archive>
+void CSGObject::cereal_save(Archive& ar) const
+{
+	for (auto it = self->map.begin(); it != self->map.end(); ++it)
+		ar(it->second);
+}
+template void CSGObject::cereal_save<cereal::JSONOutputArchive>(cereal::JSONOutputArchive& ar) const;
```

Comment:
i wouldn't do it like this, as said i think in a chat or somewhere; i would still expose the `cereal_save` with template but as well give implementations like this:
`void saveJSON(char* filename) const;`, `void saveXML(char* filename) const;` etc. and have the corresponding pairs for loading. and you yourself create there `cereal::JSONOutputArchive` etc. so this way the simple user can just choose from one of the `save*` functions, whereas the more advanced user would still have the option to use the template...